modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/CharGroupTokenizerFactoryTests.java

```diff
@@ -61,6 +64,52 @@ public void testParseTokenChars() {
         }
     }
 
+    public void testMaxTokenLength() throws IOException {
+        final Index index = new Index("test", "_na_");
+        final Settings indexSettings = newAnalysisSettingsBuilder().build();
+        IndexSettings indexProperties = IndexSettingsModule.newIndexSettings(index, indexSettings);
+        final String name = "cg";
+
+        String[] conf = new String[] {"-"};
+
+        final Settings defaultLengthSettings = newAnalysisSettingsBuilder()
+            .putList("tokenize_on_chars", conf)
+            .build();
+        CharTokenizer tokenizer = (CharTokenizer) new CharGroupTokenizerFactory(indexProperties, null, name, defaultLengthSettings).create();
```

Comment:
nit: we have a 140 chars line length limit that gets checked via "checkstyle" in CI and it complains about this line being too long. Simply splitting this is fine.